### PR TITLE
Upgrade react native version to 0.76.1

### PR DESCRIPTION
**Pull request by MasoudAlali**
As we are upgrading react-native in our main repo to 0.71.6, we should upgrade dependencies in different third-party forks we have

### CODE REVIEW

**Comment by MasoudAlali** [comment]
@FlorBosch I checked this version with Fusion and it works without any error
also we don't use `autoCompleteType` in Fusion, so there would be no problem

File: package.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
     "@types/jest": "^24.0.13",
     "@types/node": "^13.1.0",
     "@types/react-dom": "^16.8.4",
-    "@types/react-native": "^0.63.0",
+    "@types/react-native": "0.71.6",
```
Comment (FlorBosch) [inline package.json]: I believe we don't need `@types/react-native` anymore, it's now shipped with the react-native package: https://reactnative.dev/blog/2023/01/12/version-071#typescript-by-default

> We’re also offering built-in, more accurate TypeScript declarations directly from the react-native package. This means you won’t need @types/react-native any longer, and the types will be updated in lockstep with React Native releases.

**Review by FlorBosch — APPROVED** [review]
Thanks for the upgrade! 🙌

File: package.json (hunk 1 above)
Comment (MasoudAlali) [reply package.json]: Done 👍

Comment (FlorBosch) [reply package.json]: Thanks!! I'll merge it and publish the new version

**Pull request merged**